src/server/db_slice.cc

```diff
@@ -496,6 +496,14 @@ bool DbSlice::Del(DbIndex db_ind, PrimeIterator it) {
     CHECK_EQ(1u, db->mcflag.Erase(it->first));
   }
 
+  auto obj_type = it->second.ObjType();
+  if (doc_del_cb_ && (obj_type == OBJ_JSON || obj_type == OBJ_HASH)) {
+    string tmp;
+    string_view key = it->first.GetSlice(&tmp);
+    DbContext cntx{db_ind, GetCurrentTimeMs()};
+    doc_del_cb_(key, cntx, it->second);
```

Comment:
It's probably more robust to check if `doc_del_cb_` is empty before invoking it, even though we currently always set it

Reply:
1. Empty? You mean not empty? We check it in the if above
2. We only set it if search is used

Reply:
Right, my bad! Please ignore.